mailbox/api/src/main/java/org/apache/james/mailbox/acl/UnionMailboxACLResolver.java

```diff
@@ -266,28 +270,24 @@ public MailboxACL applyGlobalACL(MailboxACL resourceACL, boolean resourceOwnerIs
      * authenticated.
      */
     @Override
-    public Rfc4314Rights[] listRights(EntryKey queryKey, GroupMembershipResolver groupMembershipResolver, String resourceOwner, boolean resourceOwnerIsGroup) throws UnsupportedRightException {
+    public List<Rfc4314Rights> listRights(EntryKey queryKey, GroupMembershipResolver groupMembershipResolver, String resourceOwner, boolean resourceOwnerIsGroup) throws UnsupportedRightException {
         Rfc4314Rights[] positiveNegativePair = { MailboxACL.NO_RIGHTS, MailboxACL.NO_RIGHTS };
 
         MailboxACL userACL = resourceOwnerIsGroup ? groupGlobalACL : userGlobalACL;
         resolveRights(queryKey, groupMembershipResolver, userACL.getEntries(), resourceOwner, resourceOwnerIsGroup, positiveNegativePair);
 
         if (queryKey.isNegative()) {
-            return toListRightsArray(positiveNegativePair[NEGATIVE_INDEX]);
+            return toListRights(positiveNegativePair[NEGATIVE_INDEX]);
         } else {
-            return toListRightsArray(positiveNegativePair[POSITIVE_INDEX].except(positiveNegativePair[NEGATIVE_INDEX]));
+            return toListRights(positiveNegativePair[POSITIVE_INDEX].except(positiveNegativePair[NEGATIVE_INDEX]));
         }
     }
 
-    private static Rfc4314Rights[] toListRightsArray(Rfc4314Rights implicitRights) throws UnsupportedRightException {
-        List<Rfc4314Rights> result = new ArrayList<>();
-        result.add(implicitRights);
-        for (Right right : MailboxACL.FULL_RIGHTS.list()) {
-            if (!implicitRights.contains(right)) {
-                result.add(new Rfc4314Rights(right));
-            }
-        }
-        return result.toArray(Rfc4314Rights[]::new);
+    private static List<Rfc4314Rights> toListRights(Rfc4314Rights implicitRights) throws UnsupportedRightException {
+        return Stream.concat(
+            MailboxACL.FULL_RIGHTS.list().stream().filter(not(implicitRights::contains)).map(Rfc4314Rights::new),
```

Comment:
line breaks for readability here?

Reply:
I liked the one liner but I changed to a multiline version: is it better for you?